presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java

```diff
@@ -49,16 +48,21 @@
     @Inject
     public InternalAuthenticationManager(InternalCommunicationConfig internalCommunicationConfig, NodeInfo nodeInfo)
     {
-        this(requireNonNull(internalCommunicationConfig, "internalCommunicationConfig is null").getSharedSecret(), nodeInfo.getNodeId());
+        this(
+                requireNonNull(internalCommunicationConfig, "internalCommunicationConfig is null")
+                        .getSharedSecret()
+                        .orElse(requireNonNull(nodeInfo, "nodeInfo is null").getEnvironment()),
```

Comment:
is environment a good shared secret replacement?

Reply:
It is shared, but not a secret.  This works, but is not secure.  If you want to secure internal communication, you must set the secret.

Reply:
> If you want to secure internal communication, you must set the secret.

this i agree with.

question is -- can we have this _as a default_?

Today, when someone secures a cluster they often do this:

1. configure certs, enable HTTPS, disable HTTP
2. enable PASSWORD authenticator, usually with LDAP
3. they _realize_ workers cannot connect, so they figure they need to enable eg certs or JWT for internal comms
4. they add certs or JWT
5. everything works and they have a secure cluster

after this change, the process may look like this:

1. configure certs, enable HTTPS, disable HTTP
2. enable PASSWORD authenticator, usually with LDAP
5. everything works and they _think_ they have a secure cluster

I strongly feel this is "insecure by default" in an unobvious way.